core/java/src/org/openda/exchange/AbstractDataObject.java

```diff
@@ -0,0 +1,39 @@
+package org.openda.exchange;
+
+import org.openda.interfaces.IDataObject;
+import org.openda.interfaces.IExchangeItem;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public abstract class AbstractDataObject implements IDataObject {
+
+	public static final String[] EMPTY_STRING_ARRAY = new String[0];
+	protected Map<String, IExchangeItem> exchangeItems = new LinkedHashMap<>();
+
+	@Override
+	public String[] getExchangeItemIDs() {
+		return this.exchangeItems.keySet().toArray(EMPTY_STRING_ARRAY);
+	}
+
+	@Override
+	public String[] getExchangeItemIDs(IExchangeItem.Role role) {
+		Collection<IExchangeItem> exchangeItems = this.exchangeItems.values();
+		ArrayList<String> idsForRole = new ArrayList<>();
+		for (IExchangeItem exchangeItem : exchangeItems) {
+			IExchangeItem.Role exchangeItemRole = exchangeItem.getRole();
+			if (exchangeItemRole == role) idsForRole.add(exchangeItem.getId());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=5102854)